packages/debug/src/browser/console/debug-console-items.tsx

```diff
@@ -244,12 +240,28 @@ export class DebugVariable extends ExpressionContainer {
         const input = new SingleTextInputDialog({
             title: nls.localize('theia/debug/debugVariableInput', 'Set {0} Value', this.name),
             initialValue: this.value,
-            placeholder: nls.localizeByDefault('Value')
+            placeholder: nls.localizeByDefault('Value'),
+            validate: async (value, mode) => {
+                if (!value) {
+                    return false;
+                }
+                if (mode === 'open') {
+                    try {
+                        await this.setValue(value);
+                    } catch (error) {
+                        console.error('setValue failed:', error);
+                        if (error.body?.error) {
+                            const errorMessage: DebugProtocol.Message = error.body.error;
+                            if (errorMessage.showUser) {
+                                return formatMessage(errorMessage.format, errorMessage.variables);
+                            }
+                        }
```

Comment:
If we catch an error we should probably always return an error message, even if we didn't get one from the error. At the moment we will end up returning `true` if there is an error with an unexpected type.

Reply:
I agree, thanks for the explanation